tools/ArduinoCsCompiler/Runtime/MiniInterop.Kernel32.cs

```diff
@@ -608,6 +626,68 @@ public static UInt32 GetTimeZoneInformation(out TIME_ZONE_INFORMATION lpTimeZone
                 lpTimeZoneInformation = default;
                 return 1;
             }
+
+            private static unsafe int StrLen(byte* cstr)
+            {
+                int ret = 0;
+                while (*cstr != 0)
+                {
+                    ret++;
+                    cstr++;
+                }
+
+                return ret;
+            }
+
+            [ArduinoImplementation]
+            internal static unsafe Int32 MultiByteToWideChar(System.UInt32 CodePage, System.UInt32 dwFlags, Byte* lpMultiByteStr, Int32 cbMultiByte,
+                Char* lpWideCharStr, System.Int32 cchWideChar)
+            {
+                int bytesToConvert = cbMultiByte;
+                if (cbMultiByte == -1)
+                {
+                    bytesToConvert = StrLen(lpMultiByteStr);
+                }
+
+                if (cchWideChar == 0)
+                {
+                    return bytesToConvert + 1;
+                }
+
+                int bytesRemaining = bytesToConvert;
+                // May be -1 to run until *lpMultiByteStr is 0
+                int idx = 0;
+                while (bytesRemaining > 0)
+                {
+                    byte input = lpMultiByteStr[idx];
+                    lpWideCharStr[idx] = (Char)input;
+                    bytesRemaining--;
+                    idx++;
+                }
+
+                return bytesToConvert;
+            }
+
+            [ArduinoImplementation]
+            internal static int GetLeadByteRanges(int codePage, byte[] leadByteRanges)
+            {
+                /*
```

Comment:
Is it intentional that this method is all commented out?

Reply:
Yes. I often take the original implementation and then simplify it so that it meets the minimal requirements for our purpose. The MCU runtime doesn't support changing codepages.